main.tf

```diff
@@ -1,3 +1,10 @@
+locals {
+  cluster_encryption_config = {
+    resources         = var.cluster_encryption_config_resources
+    provider_key_arn  = var.enabled && var.enable_cluster_encryption_config && var.cluster_encryption_config_kms_key_id == "" ? join("", aws_kms_key.cluster.*.arn) : var.cluster_encryption_config_kms_key_id
```

Comment:
```suggestion
    provider_key_arn  = var.enabled && var.cluster_encryption_config_enabled && var.cluster_encryption_config_kms_key_id == "" ? join("", aws_kms_key.cluster.*.arn) : var.cluster_encryption_config_kms_key_id
```

Reply:
https://docs.cloudposse.com/terraform/terraform-best-practices/#use-positive-variable-names-to-avoid-double-negatives